core/src/test/java/org/ehcache/util/ResourcePoolsMergerTest.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.util;
+
+import org.ehcache.config.ResourcePools;
+import org.ehcache.config.ResourcePoolsBuilder;
+import org.ehcache.config.ResourceType;
+import org.ehcache.config.units.EntryUnit;
+import org.junit.Test;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author rism
+ */
+public class ResourcePoolsMergerTest {
+
+  @Test
+  public void testAddingNewTierWhileUpdating() {
+    ResourcePools existing = mock(ResourcePools.class);
+    ResourcePools toBeUpdated = mock(ResourcePools.class);
+    Set<ResourceType> existingSet = new HashSet<ResourceType>();
+    Set<ResourceType> toBeUpdatedSet = new HashSet<ResourceType>();
+    existingSet.add(ResourceType.Core.HEAP);
+    toBeUpdatedSet.add(ResourceType.Core.DISK);
+    when(existing.getResourceTypeSet()).thenReturn(existingSet);
+    when(toBeUpdated.getResourceTypeSet()).thenReturn(toBeUpdatedSet);
+    ResourcePoolMerger merger = new ResourcePoolMerger();
+    try {
+      merger.validateAndMerge(existing, toBeUpdated);
+    } catch (Exception e) {
+      assertThat(e instanceof IllegalArgumentException, is(true));
+    }
+  }
+
+  @Test
+  public void testUpdatingOffHeap() {
+    ResourcePools existing = mock(ResourcePools.class);
+    ResourcePools toBeUpdated = mock(ResourcePools.class);
+    Set<ResourceType> existingSet = new HashSet<ResourceType>();
+    Set<ResourceType> toBeUpdatedSet = new HashSet<ResourceType>();
+    existingSet.add(ResourceType.Core.HEAP);
+    existingSet.add(ResourceType.Core.OFFHEAP);
+    toBeUpdatedSet.add(ResourceType.Core.OFFHEAP);
+    when(existing.getResourceTypeSet()).thenReturn(existingSet);
+    when(toBeUpdated.getResourceTypeSet()).thenReturn(toBeUpdatedSet);
+    ResourcePoolMerger merger = new ResourcePoolMerger();
+    try {
+      merger.validateAndMerge(existing, toBeUpdated);
+    } catch (Exception e) {
+      assertThat(e instanceof UnsupportedOperationException, is(true));
```

Comment:
Catch the proper exception type directly and assert that the errors indeed relates to off heap.